src/lib/patchMigrationDocumentData.ts

```diff
@@ -0,0 +1,446 @@
+ *
+ * @returns The patched image field.
+ */
+const patchImage = (
+	image: ImageMigrationField | ImageField,
+	assets: AssetMap,
+): ImageField => {
+	if (image) {
+		if (
+			"migrationType" in image &&
+			image.migrationType === MigrationFieldType.Image
+		) {
+			// Migration image field
+			const asset = assets.get(image.id)
+			if (asset) {
+				return to.imageField(asset)
+			}
+		} else if (
+			"dimensions" in image &&
+			image.dimensions &&
+			isFilled.image(image)
+		) {
+			// Regular image field
+			const {
+				id,
+				url,
+				dimensions,
+				edit,
+				alt,
+				copyright: _,
+				...thumbnails
+			} = image
+			const asset = assets.get(id)
+
+			if (asset) {
+				const result = {
+					id: asset.id,
+					url: inheritQueryParams(asset.url, url),
+					dimensions,
+					edit,
+					alt: alt || null,
+					copyright: asset.credits || null,
+				} as ImageField<string>
+
+				if (Object.keys(thumbnails).length > 0) {
+					for (const name in thumbnails) {
+						const maybeThumbnail = (
+							thumbnails as Record<string, FilledImageFieldImage>
+						)[name]
+
+						if (is.image(maybeThumbnail)) {
+							const { url, dimensions, edit, alt } = (
+								thumbnails as Record<string, FilledImageFieldImage>
+							)[name]
+
+							result[name] = {
+								id: asset.id,
+								url: inheritQueryParams(asset.url, url),
+								dimensions,
+								edit,
+								alt: alt || null,
+								copyright: asset.credits || null,
+							}
+						}
+					}
+				}
+
+				return result
+			}
+		}
+	}
+
+	return {}
+}
+
+/**
+ * Patches references in a record of Prismic field.
+ *
+ * @typeParam TFields - Type of the record of Prismic fields.
+ * @typeParam TDocuments - Type of Prismic documents in the repository.
+ *
+ * @param record - The link field to patch.
+ * @param assets - A map of assets available in the Prismic repository.
+ * @param documents - A map of documents available in the Prismic repository.
+ *
+ * @returns The patched record.
+ */
+const patchRecord = async <
+	TFields extends Record<string, AnyRegularField | GroupField | SliceZone>,
+	TDocuments extends PrismicDocument = PrismicDocument,
+>(
+	record: FieldsToMigrationFields<TFields>,
+	assets: AssetMap,
+	documents: DocumentMap<TDocuments>,
+): Promise<TFields> => {
+	const result: Record<string, AnyRegularField | GroupField | SliceZone> = {}
+
+	for (const [key, field] of Object.entries(record)) {
+		if (is.sliceZone(field)) {
+			result[key] = await patchSliceZone(field, assets, documents)
+		} else if (is.richText(field)) {
+			result[key] = await patchRichText(field, assets, documents)
+		} else if (is.group(field)) {
+			result[key] = await patchGroup(field, assets, documents)
+		} else if (is.link(field)) {
+			result[key] = await patchLink(field, assets, documents)
+		} else if (is.image(field)) {
+			result[key] = patchImage(field, assets)
+		} else {
+			result[key] = field
+		}
+	}
```

Comment:
I don't think so because we need to patch more than migration assets and documents, all of which can have various structures depending on if they're a link, an image, a link to media, a rich text image, link, or link on image...